src/frontend/src/optimizer/plan_node/generic/except.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+
+use risingwave_common::catalog::Schema;
+
+use super::{GenericPlanNode, GenericPlanRef};
+use crate::optimizer::optimizer_context::OptimizerContextRef;
+use crate::optimizer::property::FunctionalDependencySet;
+
+/// `Except` returns the rows of its first input except any
+///  matching rows from its other inputs.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct Except<PlanRef> {
+    pub all: bool,
+    pub inputs: Vec<PlanRef>,
```

Comment:
Using `Vec` as the inputs seems not that clear compared to `Union` and `Intersect` since there's no commutative property. 🤔 (IIUC, If `inputs` is `[A, B, C]`, then this operator represents `(A - B) - C`.)

Reply:
Another approach is to split the inputs using left and others. However, currently, using inputs still seems to work well in my opinion.